app/src/main/java/com/ponkratov/weatherapp/data/mapper/CityMapper.kt

```diff
@@ -0,0 +1,51 @@
+package com.ponkratov.weatherapp.data.mapper
+
+import com.ponkratov.weatherapp.data.model.CitiesResponse
+import com.ponkratov.weatherapp.data.model.CityEntity
+import com.ponkratov.weatherapp.domain.model.City
+
+fun CitiesResponse.toDomainList(): List<City> {
+    val citiesList = mutableListOf<City>()
+    results.forEach {
+        citiesList.add(
+            City(
+                it.id ?: 0,
+                it.name ?: "",
+                it.country ?: "",
+                it.countryCode ?: "",
+                it.latitude ?: 0.0,
+                it.longitude ?: 0.0,
+                it.elevation ?: 0,
+                it.timezone ?: ""
+            )
+        )
+    }
+
+    return citiesList
```

Comment:
```suggestion
    return results.map {
            City(
                it.id ?: 0,
                it.name ?: "",
                it.country ?: "",
                it.countryCode ?: "",
                it.latitude ?: 0.0,
                it.longitude ?: 0.0,
                it.elevation ?: 0,
                it.timezone ?: ""
            )
    }
```
+ скорее всего у тебя не все параметры опциональные, если происходит парсинг dto, то примитивы при отсутствии должны заполяться дефолтными значениями